rocky/rocky/scheduler.py

```diff
@@ -189,21 +189,35 @@ class TaskNotFoundError(SchedulerError):
     message = _("Task could not be found.")
 
 
-class SchedulerClient:
+class SchedulerSession(requests.Session):
     def __init__(self, base_uri: str):
-        self.session = requests.Session()
+        super().__init__()
+        self.session = self
         self._base_uri = base_uri
-        self.is_alive()
 
+        try:
+            self.session.get(f"{self._base_uri}/health")
+        except (
+            requests.ConnectionError,
+            requests.RequestException,
+            requests.TooManyRedirects,
+            requests.ConnectTimeout,
+            requests.ReadTimeout,
+            requests.Timeout,
+        ) as error:
+            raise SchedulerError(error)
+
+
+class SchedulerClient(SchedulerSession):
```

Comment:
A client and a session are two different things. In our codebase, we usually encapsulate a session object in a client.